src/main/java/com/nike/cerberus/aws/sts/AwsStsHttpClient.java

```diff
@@ -0,0 +1,155 @@
+
+    private final OkHttpClient httpClient;
+    private final ObjectMapper objectMapper;
+
+    @Inject
+    public AwsStsHttpClient(@Named(AwsStsGuiceModule.AWS_STS_HTTP_CLIENT_NAME) final OkHttpClient httpClient,
+                            @Named(AwsStsGuiceModule.AWS_STS_OBJECT_MAPPER_NAME) final ObjectMapper objectMapper) {
+        this.httpClient = httpClient;
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * Executes the HTTP request based on the input parameters.
+     *
+     * @param headers     HTTP Headers to include in the request
+     * @param responseClass The class of the response object
+     * @return Response from the server
+     */
+    public <M> M execute(final Map<String, String> headers,
+                            final Class<M> responseClass) {
+        try {
+            Request request = buildRequest(headers);
+            Response response =  httpClient.newCall(request).execute();
+            if (response.code() >= 400 && response.code() < 500) {
+                final String msg = String.format("Failed to authenticate with AWS, error message: %s",
+                        response.body().string());
+
+                throw ApiException.newBuilder()
+                        .withApiErrors(DefaultApiError.AUTH_BAD_CREDENTIALS)
+                        .withExceptionMessage(msg)
+                        .build();
+            } else if (response.code() >= 500){
+                final String msg = String.format("Something is wrong with AWS, error message: %s",
+                        response.body().string());
+
+                throw ApiException.newBuilder()
+                        .withApiErrors(DefaultApiError.SERVICE_UNAVAILABLE)
+                        .withExceptionMessage(msg)
+                        .build();
+            }
+            return parseResponseBody(response, responseClass);
+        } catch (IOException e) {
+            throw toApiException(e);
+        }
+    }
+
+    /**
+     * Build the request
+     *
+     * @throws JsonProcessingException
+     */
+    protected Request buildRequest(Map<String, String> headers) {
+        Request.Builder requestBuilder = new Request.Builder().url(DEFAULT_AWS_STS_ENDPOINT)
+                .addHeader("Accept", DEFAULT_ACCEPTED_MEDIA_TYPE.toString());
+
+        if (headers != null) {
+            headers.forEach(requestBuilder::addHeader);
+        }
+
+        requestBuilder.addHeader("Content-Type", DEFAULT_CONTENT_MEDIA_TYPE.toString())
+                .method(DEFAULT_METHOD, RequestBody.create(DEFAULT_CONTENT_MEDIA_TYPE, DEFAULT_GET_CALLER_IDENTITY_ACTION
+                        ));
+
+        return requestBuilder.build();
+    }
+
+    protected ApiException toApiException(IOException e) {
+        if (e instanceof SSLException
+                && e.getMessage() != null
+                && e.getMessage().contains("Unrecognized SSL message, plaintext connection?")) {
+            return ApiException.newBuilder()
```

Comment:
This is only possible with local dev?  Is that right?  Should we add a comment?

Reply:
Did you ever see this exception?

Reply:
I saw this in the OneLoginHttpClient. I included it because it looks like potentially a common Exception which probably isn't the case with AWS but I kept it anyway. I can remove this.